content/blog/go-support-pulumi-2-0/index.md

```diff
@@ -0,0 +1,233 @@
+			Image:  image.ImageName,
+			Memory: pulumi.Float64(1.5),
+			Name:   pulumi.String("hello-world"),
+			Ports:  containerservice.GroupContainerPortArray{portArgs},
+		}
+		groupArgs := containerservice.GroupArgs{
+			ResourceGroupName:        resourceGroup.Name,
+			ImageRegistryCredentials: containerservice.GroupImageRegistryCredentialArray{credentialArgs},
+			OsType:                   pulumi.String("Linux"),
+			Containers:               containerservice.GroupContainerArray{containerArgs},
+			IpAddressType:            pulumi.String("public"),
+			DnsNameLabel:             pulumi.String("acigo"),
+		}
+		group, err := containerservice.NewGroup(ctx, "aci", &groupArgs)
+		if err != nil {
+			return err
+		}
+
+		ctx.Export("endpoint", group.Fqdn)
+```
+
+### Kubernetes
+
+Pulumi provides access not just to your cloud (AWS, Azure, GCP and more), but also to the whole Kubernetes API.  The Pulumi  [Kubernetes provider](https://github.com/pulumi/pulumi-kubernetes) is based on the OpenAPI specification, so is always up to date with the full functionality of the Kubernetes API. Using Pulumi to provision your Kuberentes resources gives you the declarative benefits of YAML without any of the drawbacks. Using a real language to orchestrate resources brings the benefits of strong typing, safety, and opens up options for encapsulation. Here’s a snippet from the [guestbook example](https://github.com/pulumi/examples/tree/master/kubernetes-go-guestbook):
+
+```go
+		// Redis leader Service
+		_, err = corev1.NewService(ctx, "redis-leader", &corev1.ServiceArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Name:   pulumi.String("redis-master"),
+				Labels: redisLeaderLabels,
+			},
+			Spec: &corev1.ServiceSpecArgs{
+				Ports: corev1.ServicePortArray{
+					corev1.ServicePortArgs{
+						Port:       pulumi.Int(6379),
+						TargetPort: pulumi.Int(6379),
+					},
+				},
+				Selector: redisLeaderLabels,
+			},
+		})
+		if err != nil {
+			return err
+		}
+```
+
+With the expressiveness of go, we have the power to [encapsulate services and deployments](https://github.com/pulumi/examples/blob/master/kubernetes-go-guestbook/components/serviceDeployment.go) behind higher-order components:
+
+```go
+type ServiceDeployment struct {
+	pulumi.ResourceState
+
+	FrontendIP pulumi.StringPtrOutput
+	Deployment *appsv1.Deployment
+	Service    *corev1.Service
+}
+
+type ServiceDeploymentArgs struct {
+	AllocateIPAddress pulumi.Bool
+	Image             pulumi.StringInput
+	IsMinikube        pulumi.Bool
+	Ports             pulumi.IntArrayInput
+	Replicas          pulumi.IntPtrInput
+}
+
+func NewServiceDeployment(...){...}
+```
+
+This allows us to simplify boilerplate and design strongly-typed APIs for our Kubernetes infrastructure (in this case, combining a `Service` and a `Deployment` into a single, simple, API with nice defaults for our use case):
+
+```go
+                       _, err := NewServiceDeployment(ctx, "redis-leader", &ServiceDeploymentArgs{
```

Comment:
The indentation here is still a little off if you look at the rendered site.

Reply:
The rendering for go is more indented than for other languages, but otherwise, it looks ok.

Reply:
![Screen Shot 2020-05-13 at 3 57 43 PM](https://user-images.githubusercontent.com/2553171/81874346-8d50af00-9532-11ea-9fb0-37112c1d0fc6.png)

In this instance, there may just be some extra spaces inserted. The `_, err` should have the same indentation as the `})` three lines below